### PR TITLE
Added a MTPopupWindow delegate protocol and a property "use Safari".

### PR DESCRIPTION
**Pull request by nalyd88**
The delegate protocol can be expanded but for now it defines two optional methods which are called to alert the delegate when the popup window is opened or closed.

The property "useSafari" is useful for those of us using MTPopupWindow to display local files but want to direct external links to Safari.

The test project was updated with a couple buttons to test the new features.

Thanks

Dylan

### CODE REVIEW

File: TabPopupTest/MTPopupWindow.m [hunk 1]
```diff
@@ -329,6 +339,18 @@ - (void)webView:(UIWebView *)webView didFailLoadWithError:(NSError *)error
     if (_loader) [_loader removeFromSuperview];
 }
 
+-(BOOL)webView:(UIWebView *)webView shouldStartLoadWithRequest:(NSURLRequest *)request navigationType:(UIWebViewNavigationType)navigationType;
+{
+  if (self.usesSafari) {
+    NSURL *requestURL =[request URL];
+    if (([[requestURL scheme] isEqualToString: @"http"]) && (navigationType == UIWebViewNavigationTypeLinkClicked)) {
```
Comment (icanzilb) [inline TabPopupTest/MTPopupWindow.m]: how about https links?

Comment (icanzilb) [inline TabPopupTest/MTPopupWindow.m]: use beginsWith to match both http and https protocols

Comment (nalyd88) [inline TabPopupTest/MTPopupWindow.m]: Thanks for the suggestion, I implemented it but I used "hasPrefix". I tested it to make sure it worked with both kinds of links (http and https).
Dylan

Date: Sat, 5 Jan 2013 17:00:11 -0800

File: TabPopupTest/MTPopupWindow.m [hunk 2]
```diff
@@ -216,6 +218,10 @@ -(void)showInView:(UIView*)v
 
     //animate the popup window in
     [self performSelector:@selector(animatePopup:) withObject:v afterDelay:0.01];
+  
+    // Attempt to alert the delegate.
+    if ([_delegate respondsToSelector:@selector(didShowMTPopupWindow:)])
+      [_delegate didShowMTPopupWindow:self];
```
Comment (icanzilb) [inline TabPopupTest/MTPopupWindow.m]: If you want to call the delegate method at this point it has to be named "willShowMTPopupWindow" - if you want to actually have a "didShowMTPopupWindow" delegate method it should be invoked only after the animatePopup: method finishes the animation. Besides this I think all the rest of the changes are cool

**Comment by icanzilb** [comment]
All looks good. Thanks for contributing!

**Comment by icanzilb** [comment]
@nalyd88 http://www.touch-code-magazine.com/mtpopupwindow-is-new-delegate-protocol/

**Pull request merged**